loris/transforms.py

```diff
@@ -131,42 +184,34 @@ def _derive_with_pil(self, im, target_fp, image_request, rotate=True, crop=True)
             im.save(target_fp, quality=90)
 
 
-class _PillowTransformer(_AbstractTransformer):
-    def __init__(self, config):
-        super(_PillowTransformer, self).__init__(config)
-
+class _PillowTransformer(_AbstractTransformer, _MapProfilesToSRGBMixin):
     def transform(self, src_fp, target_fp, image_request):
         im = Image.open(src_fp)
         self._derive_with_pil(im, target_fp, image_request)
 
+
 class JPG_Transformer(_PillowTransformer):
-    def __init__(self, config): super(JPG_Transformer, self).__init__(config)
+    pass
+
 
 class TIF_Transformer(_PillowTransformer):
-    def __init__(self, config): super(TIF_Transformer, self).__init__(config)
+    pass
+
 
 class PNG_Transformer(_PillowTransformer):
-    def __init__(self, config): super(PNG_Transformer, self).__init__(config)
+    pass
 
-class _AbstractJP2Transformer(_AbstractTransformer):
+
+class _AbstractJP2Transformer(_AbstractTransformer, _MapProfilesToSRGBMixin):
     '''
     Shared methods and configuration for the Kakadu and OpenJPEG transformers.
 
     Exits if OSError is raised during init.
     '''
     def __init__(self, config):
-        self.map_profile_to_srgb = bool(config['map_profile_to_srgb'])
         self.mkfifo = config['mkfifo']
         self.tmp_dp = config['tmp_dp']
 
-        if self.map_profile_to_srgb and \
-            ('PIL.ImageCms' not in sys.modules and 'ImageCms' not in sys.modules):
-            logger.warn('Could not import profileToProfile from ImageCms.')
-            logger.warn('Images will not have their embedded color profiles mapped to sSRGB.')
```

Comment:
I think this warning was unhittable – because `profileToProfile` is imported at the top of this file, if the libraries aren’t installed, we’d never reach this point in the file.